src/main/java/org/jitsi/videobridge/cc/SimulcastController.java

```diff
@@ -1099,14 +1210,81 @@ boolean accept(FrameDesc source, byte[] buf, int off, int len)
 
                     if (tsTranslation != null)
                     {
-                        long srcTs = pktOut.getTimestamp();
                         long dstTs = tsTranslation.apply(srcTs);
 
                         if (dstTs != srcTs)
                         {
                             pktOut.setTimestamp(dstTs);
                         }
                     }
+
+                    if (dstPictureID > -1)
+                    {
+                        MediaStreamTrackDesc source = weakSource.get();
+                        assert source != null;
+
+                        REDBlock redBlock = source.getMediaStreamTrackReceiver()
+                            .getStream().getPayloadBlock(
+                                pktOut.getBuffer(),
+                                pktOut.getOffset(),
+                                pktOut.getLength());
+
+                        if (!DePacketizer
+                            .VP8PayloadDescriptor.hasExtendedPictureId(
+                            redBlock.getBuffer(), redBlock.getOffset(),
+                            redBlock.getLength()))
+                        {
+                            // XXX we have observed that using a non-extended
```

Comment:
Correction (due to George): this does not execute on every packet, only on ones without an extended PictureID. Ignore the comment.